docs/CONFIG-VARS.md

```diff
@@ -345,5 +345,45 @@ The nfs-client is currently supported by the newer nfs-subdir-external-provision
 | V4MT_PROVIDER_PASSWORD_{{TENANT-ID}} | Optional: Unique sasprovider password for each tenant being onboarded. {{TENANT-ID}} must be in uppercase | string | | false | Maps to SAS_PROVIDER_PASSWORD_{{TENANT-ID}}. When V4MT_PROVIDER_PASSWORD_{{TENANT-ID}} is specified V4MT_PROVIDER_PASSWORD can not be used. See details [here](https://go.documentation.sas.com/doc/en/itopscdc/default/caltenants/p0emzq13c0zbhxn1hktsdlmig934.htm#p1ghvmezrb3cvxn1h7vg4uguqct6) | multi-tenancy |
-| V4MT_CAS_WORKER_COUNT | The number of CAS worker nodes for tenants. Default is 0 (SMP) | int | 0 | false | | multi-tenancy |
-| V4MT_CAS_BACKUP | Set this flag to 1 to include a CAS backup controller | int | Disabled by default | false | | multi-tenancy |
\ No newline at end of file
+| V4MT_TENANT_CAS_CUSTOMIZATION | Map of objects with all tenant CAS customization variables. See the format below | | | false | | multi-tenancy |
+
+### Tenant CAS Customization
+
+Some of the tenant CAS customizations can be defined with the V4MT_TENANT_CAS_CUSTOMIZATION variable which is a map of objects. The variable has the following format:
+
+```bash
+V4MT_TENANT_CAS_CUSTOMIZATION:
+  <TENANT-ID1>:
+    ...
+  <TENANT-ID2>:
+    ...
+  ...
+```
+Below is the list of parameters each element can contain.
+
+| Name | Description | Type | Default | Required | Notes | Tasks |
+| :--- | ---: | ---: | ---: | ---: | ---: | ---: |
+| V4MT_CFG_CAS_RAM | Amount of ram to allocate to per CAS node | string | | false | Numeric value followed by the units, such as 32Gi for 32 gigabytes. In Kubernetes, the units for gigabytes is Gi. | multi-tenancy |
+| V4MT_CFG_CAS_CORES | Amount of cpu cores to allocate per CAS node | string | | false | Either a whole number, representing that number of cores, or a number followed by m, indicating that number of milli-cores. | multi-tenancy |
+| V4MT_CFG_CAS_ENABLE_LOADBALANCER | Setup LB to access CAS binary ports | bool | false | false | | multi-tenancy |
+| V4MT_LOADBALANCER_SOURCE_RANGES | Loadbalancer source ranges specific to the tenant | list | false | false | | multi-tenancy |
+| V4MT_CFG_CAS_WORKER_COUNT | The number of CAS worker nodes for tenants. Default is 0 (SMP) | int | 0 | false | | multi-tenancy |
+| V4MT_CFG_CAS_ENABLE_BACKUP_CONTROLLER | Enable backup cas controller | bool | false | false | | multi-tenancy |
+
+Example:
+
+```bash
+V4MT_TENANT_CAS_CUSTOMIZATION:
+  ACME:
+    V4MT_CFG_CAS_RAM: 3Gi
+    V4MT_CFG_CAS_CORES: 300m
+    V4MT_CFG_CAS_ENABLE_LOADBALANCER: true
+    V4MT_LOADBALANCER_SOURCE_RANGES: ['0.0.0.0/0']
+    V4MT_CFG_CAS_WORKER_COUNT: 0
+    V4MT_CFG_CAS_ENABLE_BACKUP_CONTROLLER: false
```

Comment:
These are not V4MT varaibles at this point the customer is simply applying those so they should look like the following:

```bash
V4MT_TENANT_CAS_CUSTOMIZATION:
  acme:
    cpus: 4
    memory: 8
    loadbalancer_enabled: true
    loadbalancer_source_ranges: ['0.0.0.0/0']
    backup_controller_enabled: false
    worker_count: 0
```
Then in the code we map things over to whatever we need internally. This is much cleaner for the end user to understand if that's possible given the complexity of these variables. You'll need to update the descriptions to let folks know if they are assigning actual CPUs, memory, etc or if those are system resource requests from kubernetes which is very different.

Also wondering about the load balancer, is that setup by namespace or do all these tenants live in the same namespace where traffic can cross?

Reply:
I'll update the variables to match the structure you mentioned.
All the tenants live in the same namespace as Viya. From the CAS documentation looks like you could use the cas-enable-external-services.yaml to set different loadbalancer source ranges for different CAS servers. In the current setup I could only verify that the values that are specified got assigned correctly. Not sure, maybe testing team could do intensive test around this.

Reply:
About description, these variables under the exactly same as on the provider side with same description. The only difference is they are for tenants. See the original provider CAS variables here:
https://github.com/sassoftware/viya4-deployment/blob/staging/docs/CONFIG-VARS.md#cas
Let me if we still need any further updates on descriptions.

Reply:
I agree with the description. Those are fine. We've already been telling folks the same info.